python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -605,3 +622,298 @@ def makeDataRefList(self, namespace):
             )
             self.refList.append(dataRef)
 
+def countMaskFromFootprint(mask, footprint, bitmask, ignoreMask):
```

Comment:
You can also write this as

    def countMaskFromFootprint(mask, footprint, bitmask, ignoreMask):
        footprint = afwDetect.Footprint(footprint)   # makes a copy
        footprint.intersect(mask, bitmask)
        footprint.intersect(mask, ~ignoreMask)
        return footprint.getNpix()

which is C++ and doesn't require a temp Mask (it makes a temp Footprint instead -- swings and roundabouts)